nixos/modules/security/acme/default.nix

```diff
@@ -222,6 +222,8 @@ let
       ++ optionals (data.dnsResolver != null) [ "--dns.resolvers" data.dnsResolver ]
     ) else if data.s3Bucket != null then [ "--http" "--http.s3-bucket" data.s3Bucket ]
     else if data.listenHTTP != null then [ "--http" "--http.port" data.listenHTTP ]
+    # TODO: why is defaults.tlsMode not applied to data.tlsMode so we have to do this ||?
```

Comment:
I'd like to see this resolved. I don't know the module well enough to know what's going on here myself though.

Reply:
Figured it out. My options were missing that `defaultAndText` thing.